ToolBox/BovespaDownloader/BovespaDataDownloader.cs

```diff
@@ -0,0 +1,449 @@
+                                {
+                                    yield return new Tick
+                                    {
+                                        Symbol = symbol,
+                                        TickType = tickType,
+                                        Time = DateTime.ParseExact(csv[0], "yyyy-MM-dd", null).Add(TimeSpan.Parse(csv[5])),
+                                        Value = decimal.Parse(csv[3], NumberStyles.Any, CultureInfo.InvariantCulture),
+                                        Quantity = int.Parse(csv[4])
+                                    };
+                                }
+
+                                // For QuoteBar, we want best buy and sell quote, 
+                                // therefore Priority indicator (csv[7]) ought to be zero. 
+                                if (long.Parse(csv[7]) > 0) continue;
+
+                                if (isBid)
+                                {
+                                    yield return new Tick
+                                    {
+                                        Symbol = symbol,
+                                        TickType = tickType,
+                                        Time = DateTime.ParseExact(csv[0], "yyyy-MM-dd", null).Add(TimeSpan.Parse(csv[6])),
+                                        BidPrice = decimal.Parse(csv[8], NumberStyles.Any, CultureInfo.InvariantCulture),
+                                        BidSize = int.Parse(csv[9])
+                                    };
+                                }
+
+                                if (isAsk)
+                                {
+                                    yield return new Tick
+                                    {
+                                        Symbol = symbol,
+                                        TickType = tickType,
+                                        Time = DateTime.ParseExact(csv[0], "yyyy-MM-dd", null).Add(TimeSpan.Parse(csv[6])),
+                                        AskPrice = decimal.Parse(csv[8], NumberStyles.Any, CultureInfo.InvariantCulture),
+                                        AskSize = int.Parse(csv[9])
+                                    };
+                                }
+                            }
+                        }
+                    }
+                }
+            }
+        }
+
+        /// <summary>
+        /// Aggregates a list of ticks at the requested resolution
+        /// </summary>
+        /// <param name="symbol">Symbol</param>
+        /// <param name="ticks">Input tick data</param>
+        /// <param name="resolution">Output resolution</param>
+        /// <returns>Enumerable of trade bars or quote bars</returns>
+        private static IEnumerable<BaseData> AggregateTicks(Symbol symbol, IEnumerable<BaseData> data, Resolution resolution)
+        {
+            var interval = new Dictionary<Resolution, TimeSpan>()
```

Comment:
The resolution has a nifty [extension method](https://github.com/QuantConnect/Lean/blob/master/Common/Extensions.cs#L518) defined, `ToTimeSpan` that will do this for you :smile: